.github/workflows/deploy.yml

```diff
@@ -36,15 +36,11 @@ jobs:
       - name: Cache local Maven repository
         uses: actions/cache@v2
         with:
-          path: ~/.m2/repository
-          key: ${{ runner.os }}-${{ matrix.version }}-maven-${{ hashFiles('**/pom.xml') }}
-          restore-keys: ${{ runner.os }}-${{ matrix.project }}-maven-
-      - name: Cache Docker
-        uses: actions/cache@v2
-        with:
-          path: ~/.docker
-          key: ${{ runner.os }}-${{ matrix.version }}-docker-${{ hashFiles('**/Dockerfile') }}
-          restore-keys: ${{ runner.os }}-${{ matrix.project }}-docker
+          path: ./m2repository  # Shared with the Docker build context via .dockerignore
+          key: ${{ runner.os }}-${{ matrix.version }}-maven-${{ hashFiles('parent-pom.xml', format('{0}/pom.xml', matrix.version)) }}
```

Comment:
fyi @openzipkin/devops-tooling if you ever wanted to know.. nested variables aren't supported in GH actions. You have to play games that feel like awk :D